File: back-end/src/main/java/CRM4RTONOCTURNOA/CRM/service/UsuarioService.java
package CRM4RTONOCTURNOA.CRM.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import CRM4RTONOCTURNOA.CRM.entity.Usuario;
import CRM4RTONOCTURNOA.CRM.repository.UsuarioRepository;

@Service
public class UsuarioService {

    @Autowired
    UsuarioRepository usuarioRepository;

    //Create y Update
    public Usuario save(Usuario usuario)
    {
      return usuarioRepository.save(usuario);
    }
    //Read
    public Usuario findById(Long id)
    {
        return usuarioRepository.findById(id).get();
    }

    //Delete
    public void deleteById(Long id)
    {
        usuarioRepository.deleteById(id);
    }

    public List<Usuario> findAll()
    {
        return usuarioRepository.findAll();
    }
    
}

File: fron-end/src/app/usuario/usuario.service.ts
import { Injectable } from '@angular/core';
import { Usuario } from './usuario';
import { Observable } from 'rxjs';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root',
})
export class UsuarioService {
  private httpOptions = {
    headers: new HttpHeaders({ 'Content-Type': 'application/json' }),
  };

  private url = environment.API_URL + '/usuario';

  constructor(private http: HttpClient) {}

  //Create
  public save(usuario: Usuario): Observable<Usuario> {
    return this.http.post<Usuario>(
      this.url + '/save',
      usuario,
      this.httpOptions
    );
  }

  //Read
  public findById(id: number): Observable<Usuario> {
    return this.http.get<Usuario>(this.url + '/id', this.httpOptions);
  }

  //Update
  public update(usuario: Usuario): Observable<Usuario> {
    return this.http.put<Usuario>(
      this.url + '/update',
      usuario,
      this.httpOptions
    );
  }

  //Delete
  public deleteById(id: number): Observable<Usuario> {
    return this.http.delete<Usuario>(
      this.url + '/deleteById/' + id,
      this.httpOptions
    );
  }

  //FinalAll
  public findAll(): Observable<Usuario[]> {
    return this.http.get<Usuario[]>(this.url + '/findAll', this.httpOptions);
  }
}
